tools/scripts/script.ps1

```diff
@@ -0,0 +1,204 @@
+            [Environment]::SetEnvironmentVariable('Path', $newUserPath, [EnvironmentVariableTarget]::User)
+            Write-Host "[AIShell added to Path]" -ForegroundColor Green
+        }
+    } else {
+        sudo tar -xzf yourfile.tar.gz -C $destination
+        if ($LASTEXITCODE -ne 0) {
+            throw "Failed to extract '$tempPath' to the folder '$destination'."
+        }
+
+        $aishPath = Join-Path $destination 'aish'
+        sudo chmod +x $aishPath
+        if ($LASTEXITCODE -ne 0) {
+            throw "Failed to set the execution permission to the executable '$aishPath'."
+        }
+
+        # No need to setup the Path env variable as the symbolic link is already within Path.
+        $symlink = $Script:MacSymbolicLink
+        sudo ln -s $aishPath $symlink
+        if ($LASTEXITCODE -ne 0) {
+            throw "Failed to create the symbolic link '$symlink' to '$aishPath'."
+        }
+    }
+}
+
+function Uninstall-AIShellApp {
+    $destination = $Script:InstallLocation
+    if (Test-Path $destination) {
+        if ($IsWindows) {
+            Remove-Item -Path $destination -Recurse -Force -ErrorAction Stop
+
+            # Update the user-scope Path env variables to remove AIShell.
+            $userPath = [Environment]::GetEnvironmentVariable('Path', [EnvironmentVariableTarget]::User)
+            if ($userPath.Contains($destination)) {
+                Write-Host "[Removing AIShell from user-scope Path]" -ForegroundColor Green
+                $newUserPath = $userPath.Split(';', [StringSplitOptions]::RemoveEmptyEntries -bor [StringSplitOptions]::TrimEntries) |
+                    Where-Object { $_ -ne $destination } |
+                    Join-String -Separator ';'
+                [Environment]::SetEnvironmentVariable("Path", $newUserPath, [EnvironmentVariableTarget]::User)
+                Write-Host "[AIShell removed from user-scope PATH]" -ForegroundColor Green
+            }
+        } else {
+            sudo rm -rf $destination
+            if ($LASTEXITCODE -ne 0) {
+                throw "Failed to remove the AIShell app from '$destination'."
+            }
+
+            $symlink = $Script:MacSymbolicLink
+            sudo rm $symlink
+            if ($LASTEXITCODE -ne 0) {
+                throw "Failed to remove the symbolic link '$symlink'."
+            }
+        }
+        Write-Host "[AIShell removed from '$destination']" -ForegroundColor Green
+    } else {
+        Write-Host "[AIShell cannot be found at '$destination', skip removing]" -ForegroundColor Yellow
+    }
+}
+
+function Install-AIShellModule {
+    Write-Host "[Installing the PowerShell module 'AIShell' ...]`n" -ForegroundColor Green
+    Install-PSResource -Name AIShell -Repository PSGallery -Prerelease -ErrorAction Stop
+    Write-Host "[AIShell Module installed]" -ForegroundColor Green
+}
+
+function Uninstall-AIShellModule {
```

Comment:
This only works if I have not previously ran Start-AIShell in my session. Unsure if there is a work around, if not possible then we can improve the error message to be "This PowerShell session currently has AIShell imported and cannot uninstall it, please restart this session and try uninstalling again"

Reply:
This seems expected and the current error message is clear enough for actions